signac/contrib/project.py

```diff
@@ -156,6 +156,15 @@ def __init__(self, config=None, _ignore_schema_version=False):
         self._fn_doc = os.path.join(self._rd, self.FN_DOCUMENT)
         self._document = None
 
+        # Prepare Workspace Directory
+        if not os.path.isdir(self._wd):
+            try:
+                _mkdir_p(self._wd)
+            except OSError:
+                logger.error(
+                    "Error occurred while trying to create "
+                    "workspace directory for project {}".format(self.id))
```

Comment:
After calling `logger.error(...)`, the error still needs to be raised because the project initialization has failed. Just add the statement `raise` after this error is logged.

Adding `raise` would also match the current behavior in the Job class when attempting to create the workspace directory.
https://github.com/glotzerlab/signac/blob/dc870ac06159cae97e1475894913d1def567555e/signac/contrib/job.py#L332-L341

Reply:
Add a period at the end of the error message.
```suggestion
                    "workspace directory for project {}.".format(self.id))
```